### PR TITLE
Fix ROS2 python tests

### LINKED ISSUE #574: Issue with missing typeguard and pyYaml in bundled python.
There were some complaints during the initial project configuration.
I've run a simple script to configure the project:
```
#! /bin/sh
export O3DE_HOME=/home/robotec/o3de_ws/o3de/
export O3DE_EXTRAS_HOME=/home/robotec/o3de_ws/o3de-extras/
export PROJECT_NAME=Ros2Project
export PROJECT_PATH=${HOME}/projects/${PROJECT_NAME}
${O3DE_HOME}/scripts/o3de.sh create-project --project-path $PROJECT_PATH --template-path ${O3DE_EXTRAS_HOME}/Templates/Ros2ProjectTemplate -f

cd $PROJECT_PATH
cmake -B build/linux -G "Ninja Multi-Config" -DLY_DISABLE_TEST_MODULES=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DLY_STRIP_DEBUG_SYMBOLS=ON -DLY_UNITY_BUILD=OFF -DAZ_USE_PHYSX5:BOOL=ON 
cmake --build build/linux --config profile --target ${PROJECT_NAME} Editor ${PROJECT_NAME}.Assets 

${PROJECT_PATH}/build/linux/bin/profile/Editor
```
I got warning from bundled Python:
```
-- Using package /home/robotec/.o3de/3rdParty/packages/pyside2-5.15.2.1-py3.10-rev6-linux
-- Python: linking /home/robotec/.o3de/3rdParty/packages/pyside2-5.15.2.1-py3.10-rev6-linux/pyside2/lib/python3.10/site-packages into python site-packages...
-- Python: Getting/Checking packages listed in /home/robotec/o3de_ws/o3de/Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/requirements.txt
-- Python: Getting/Checking packages listed in /home/robotec/o3de_ws/o3de/Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/requirements.txt - New packages were installed
CMake Warning at /home/robotec/o3de_ws/o3de/cmake/LYPython.cmake:148 (message):
  PIP reports unmet dependencies: generate-parameter-library-py 0.3.6
  requires pyyaml, which is not installed.

  generate-parameter-library-py 0.3.6 requires typeguard, which is not
  installed.

  launch-ros 0.19.6 requires pyyaml, which is not installed.

Call Stack (most recent call first):
  /home/robotec/o3de_ws/o3de/Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/CMakeLists.txt:8 (update_pip_requirements)
```
The full log is here : https://gist.github.com/michalpelka/3a59e313d055513ab3449bb8e190404f
Branch stabilization/2310.

### LINKED ISSUE #575: Python tests do not work anymore.
On the stabilization/2310 branch, the Python test (at this moment only body to run ROS2 integration tests) fails to initialize.

From what I can see in the log, we tries to run `rospy` against o3de's bundled Python, and pyyaml is missing:
```
144:   File "/opt/ros/humble/lib/python3.10/site-packages/launch/utilities/type_utils.py", line 29, in <module>
144:     import yaml
144: ModuleNotFoundError: No module named 'yaml'
```

Full log:
https://gist.github.com/michalpelka/acebb8b1d1d6807a17acaa63bae415f5

### PR DESCRIPTION
**Pull request by MateuszWasilewski**
## What does this PR do?
fixes #575 
resolves #574 

Adds pip dependencies (connected to ros2) to `o3de/python` if ros2 gem is enabled.

## How was this PR tested?

On non headless device run:

```bash
cd <o3de>

scripts/o3de.sh enable-gem --gem-path ../o3de-extras/Gems/ROS2 --project-name AutomatedTesting
# Build all tests for AutomatedTesting project
python/python.sh -u scripts/build/ci_build.py --platform Linux --type profile_nounity

cd build/linux
ctest -C profile -R "ROS2.PythonTests" --output-on-failure
```

On o3de commit `5ae4e51252`

### CODE REVIEW

File: Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py [hunk 1]
```diff
@@ -66,9 +66,13 @@ def SmokeTest_EnterGameModeWorks():
     # Required for automated tests
     helper.init_idle()
 
-    # Open the level called "Warehouse".
-    # We use a warehouse level for a smoke test - it already has a robot prefab present
-    helper.open_level(level="Warehouse", directory='')
+    # Open the level called "DefaultLevel".
+    # We use a DefaultLevel level for a smoke test.
+    # ROS2 System Component should publish below topics regardless of level
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py:71]:
```suggestion
    # ROS2 System Component should publish topics listed below regardless of level
```

**Review by jhanca-robotecai — COMMENTED** [review]
Tested on o3de `cec608023e`

Tests passed correctly after changing the default level name - my project did not have `DefaultLevel` level.

```diff
diff --git a/Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py b/Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py
index e97d406b..7c5c728b 100644
--- a/Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py
+++ b/Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py
@@ -72,7 +72,7 @@ def SmokeTest_EnterGameModeWorks():
     # - /tf
     # - /tf_static
     # - /clock
-    helper.open_level(level="DefaultLevel", directory='')
+    helper.open_level(level="DemoLevel", directory='')
 
     topics_before_game_mode = check_topics()
```

File: Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py [hunk 2]
```diff
@@ -66,9 +66,13 @@ def SmokeTest_EnterGameModeWorks():
     # Required for automated tests
     helper.init_idle()
 
-    # Open the level called "Warehouse".
-    # We use a warehouse level for a smoke test - it already has a robot prefab present
-    helper.open_level(level="Warehouse", directory='')
+    # Open the level called "DefaultLevel".
+    # We use a DefaultLevel level for a smoke test.
+    # ROS2 System Component should publish topics listed below regardless of level
```
Comment (pijaro) [inline Gems/ROS2/Code/PythonTests/tests/SmokeTests_EnterGameModeWorks.py:71]: Nit: `ROS2SystemComponent` if referring to the "ROS2" gem component, or `ROS 2 system component` if referring to ROS 2 as a general behavior.

**Review by pijaro — APPROVED** [review]
Approved with a minor nitpick.

**Pull request merged**